sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_ieee802154.c

```diff
@@ -113,6 +113,21 @@ static gnrc_pktsnip_t *_recv(gnrc_netdev2_t *gnrc_netdev2)
                 return NULL;
             }
             hdr = netif_hdr->data;
+
+            /* XXX: workaround for drivers that hear themselves */
+            uint8_t *tmp_src;
+            tmp_src = gnrc_netif_hdr_get_src_addr(hdr);
+
+            uint8_t my_long_addr[IEEE802154_LONG_ADDRESS_LEN];
+            netdev2_ieee802154_get(state, NETOPT_ADDRESS_LONG, my_long_addr, sizeof(my_long_addr));
```

Comment:
I may have some lack of knowledge regarding the netdev2 api, but which address would this call return if I have **two** 15.4 devices attached to my board? Shouldn't you somehow specify that you want the address from the incoming netif header in this case?